tests/unit/s2n_certificate_test.c

```diff
@@ -804,5 +804,41 @@ int main(int argc, char **argv)
         EXPECT_BYTEARRAY_EQUAL(chain->sct_list.data, sct_list, sizeof(sct_list));
     };
 
+    /* Test s2n_cert_chain_and_key_load_cns */
+    {
+        DEFER_CLEANUP(struct s2n_cert_chain_and_key *chain = NULL,
+                s2n_cert_chain_and_key_ptr_free);
+        EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain,
+                S2N_RSA_2048_SHA256_INTERMEDIATE_CERT_CUSTOM_OID,
+                S2N_RSA_2048_SHA256_INTERMEDIATE_CA_KEY));
+        POSIX_ENSURE_REF(chain->cert_chain);
+        POSIX_ENSURE_REF(chain->cert_chain->head);
+        struct s2n_cert *head = chain->cert_chain->head;
+        struct s2n_blob *leaf_bytes = &head->raw;
+        EXPECT_NOT_NULL(leaf_bytes);
+        const unsigned char *leaf_der = leaf_bytes->data;
+        DEFER_CLEANUP(X509 *cert = NULL, X509_free_pointer);
+        cert = d2i_X509(NULL, &leaf_der, leaf_bytes->size);
+        X509_NAME *x509_name = X509_NAME_new();
```

Comment:
This is a pretty dense chunk of code. I'd consider breaking it up a bit. I'm also not sure your null checks are quite right.
```suggestion
        POSIX_ENSURE_REF(chain->cert_chain);
        
        struct s2n_cert *head = chain->cert_chain->head;
        POSIX_ENSURE_REF(head);
        
        struct s2n_blob *leaf_bytes = &head->raw;
        const unsigned char *leaf_der = leaf_bytes->data;
        EXPECT_NOT_NULL(leaf_der);
        
        DEFER_CLEANUP(X509 *cert = NULL, X509_free_pointer);
        cert = d2i_X509(NULL, &leaf_der, leaf_bytes->size);
        EXPECT_NOT_NULL(cert);
        
        X509_NAME *x509_name = X509_NAME_new();
        EXPECT_NOT_NULL(x509_name);
        
```